### PR TITLE
port Windows driver support to the next branch

### PR DESCRIPTION
**Pull request by tandasat**
### Changes
This gigantic change is basically port of Windows driver support from the master branch. Namely,
- deletion of `myinttypes.h`
- documents updates
- suppression of warnings (integer truncation, variable name shadowing, etc)
- addition of contrib/cs_driver/, windows/, msvc/capstone_static_winkernel/ and msvc/test_winkernel/
- make use of `CAPSTONE_API`

We would be better of asking an author of WinCE support since I have made non-trivial changes in `platform.h` to clean up #ifdefs with regard to `stdint.h` inclusion and was unable to test for WinCE.

See attached and compare test log results in [test_logs.zip](https://github.com/aquynh/capstone/files/261210/test_logs.zip) from both the user-mode ([code_test_all.zip](https://github.com/aquynh/capstone/files/261211/code_test_all.zip)) and the kernel-mode (test_winkernel) to verify if this porting is healthy in functionality wise.

### Issue
I found an issue that Windows kernel did not support formatting string for any of the floating point types at alI, and found 4 instances in M68K test code. See the attached logs. 

As for an impact of this issue, those are code likely to be affected on the 'next' branch: 
- capstone\arch\AArch64\AArch64InstPrinter.c (1 hit)
 - 	Line 1278: 	SStream_concat(O, "#%.8f", FPImm);
- capstone\arch\M68K\M68KInstPrinter.c (2 hits)
 - 	Line 151: 					 SStream_concat(O, "#%f", op->simm);
 - 	Line 153: 					 SStream_concat(O, "#%f", op->dimm);
- capstone\suite\arm\test_arm_regression.c (1 hit)
 - 	Line 83: 				_this_printf("\t\toperands[%u].type: FP = %f\n", i, op->fp);
- capstone\suite\benchmark\test_iter_benchmark.c (1 hit)
 - 	Line 92: 	printf("time used:%f\n", timeUsed);
- capstone\tests\test_arm.c (2 hits)
 - 	Line 61: 				printf("\t\toperands[%u].type: FP = %f\n", i, op->fp);
 - 	Line 76: 					printf("\t\t\toperands[%u].mem.lshift: 0x%x\n", i, op->mem.lshift);
- capstone\tests\test_arm64.c (1 hit)
 - 	Line 59: 				printf("\t\toperands[%u].type: FP = %f\n", i, op->fp);
- capstone\tests\test_m68k.c (2 hits)
 - 	Line 90: 						printf("\t\toperands[%u].type: IMM = %f\n", i, op->simm);
 - 	Line 92: 						printf("\t\toperands[%u].type: IMM = %lf\n", i, op->dimm);
(grep: %[^\s]*f)

See #681 for more details.

### CODE REVIEW

**Comment by aquynh** [comment]
thank you for yet another high quality PR!

regarding the %f, this means the "master" branch also suffers from this issue?

**Comment by aquynh** [comment]
i merged this into a new branch "windrv" at https://github.com/aquynh/capstone/branches

@koutheir: this new branch made some changes that may affect WindowsCE. could you please test it?

**Comment by tandasat** [comment]
On the floating pointer issue, yes, it seems. I found those code in the master branch, although I have not confirmed exactly how those code would be exercised.  

- \capstone_master\arch\AArch64\AArch64InstPrinter.c (1 hit)
 - 	Line 1040: 	SStream_concat(O, "#%.8f", FPImm);
- \capstone_master\arch\ARM\ARMInstPrinter.c (1 hit)
 - 	Line 2106: 	SStream_concat(O, "#%e", getFPImmFloat((unsigned int)MCOperand_getImm(MO)));

`Search "%[^\s]*[feEgGaA]" (108 hits in 21 files)`

Given that it is likely to affect master, I will create an issue to make discussion easier.

**Comment by tandasat** [comment]
I have filed #681 so that we can discuss it in a separate place.

**Comment by aquynh** [comment]
merged your latest commit to "windrv" branch, thanks.

**Comment by tandasat** [comment]
Almost a week past since we asked @koutheir. Should we proceed review and merge this changes?

**Comment by koutheir** [comment]
You can merge if you like, and I will try to test the changes as soon as I can.

**Comment by tandasat** [comment]
Thanks @koutheir.  

@aquynh, let me know if you find any code to update before merging it.

**Comment by aquynh** [comment]
Ok, will do a final review before merging it, thanks.

File: tests/test_arm64.c [hunk 1]
```diff
@@ -91,7 +96,7 @@ static void print_insn_detail(cs_insn *ins)
 				break;
 		}
 
-		uint8_t access = op->access;
+		uint32_t access = op->access;
```
Comment (aquynh) [inline tests/test_arm64.c]: why did you change this type from uint8_t to uint32_t?

Comment (tandasat) [inline tests/test_arm64.c]: I thought it caused a warning regarding integer-truncation. But I cannot reproduce it now. It may be my mistake or may be other changes indirectly addressed the warning. I will revert it to uint8_t shortly

Comment (aquynh) [inline tests/test_arm64.c]: no worry, i will fix that.

**Comment by aquynh** [comment]
merged into the "next" branch, thanks for your amazing work!

**Comment by tandasat** [comment]
Thank you for your very responsive reply and help! Contribution experience was awesome to me.

I will make progress for #686 if he does not respond within few days.

**Comment by stonedreamforest** [comment]
:+1:
